### ISSUE TITLE
Readable Code

### ISSUE THREAD

**Issue by Rekyt** [adrientaudiere/lottery#2]
Hey @frmunoz @adrientaudiere @pierredenelle, we should clean up commented code in files and add more comment to the rest of the functions to make it more user-friendly.
For example we still have this section in https://github.com/adrientaudiere/lottery/blob/245939fc105e0bdf8d18232b6e3493a514d3eb40/pkg/R/forward.R  (lines 49-52):
```r
#else if(is.vector(initial)) {
  #  J <- sum(initial)
  #  a <- data.frame(id = paste("init", 1:J, sep = ""), sp = unlist(sapply(sp.names, function(x) rep(x, initial[x]))), trait = rep(NA, J), stringsAsFactors = F)
  #}
```

And there is a single comment (!) for `coalesc()` function in https://github.com/adrientaudiere/lottery/blob/245939fc105e0bdf8d18232b6e3493a514d3eb40/pkg/R/coalesc.R

We could comment more and drop any commented "debug code"

**Comment by Rekyt** [comment]
I started a new branch to add comments adrientaudiere/lottery/tree/code_comment 2da9437
and also to rename variables

**Comment by Rekyt** [comment]
Fixed this with 17f4a1a3941e49bc6c190a375b4700e8360e56cb
